### PR TITLE
Feature/employee month report

### PR DESCRIPTION
**Pull request by vladPinteaFrontmen**
# Changes
New button on week view lets you navigate to monthly overview

## Related issues
https://github.com/FrontMen/fm-hours/issues/151

## Added
Weekly overview page:
* can filter by billable or all
* can filter by customer
* can navigate back to weekly view
* can trigger a brwoser print of the selected records
* can change the month to be viewed, up until the current month

## Changed
New button on week view lets you navigate to monthly overview

## How to test
1. go to week view
2. click on monthly overview - shows the current month, not the month corresponding to the week you came from
3. filter the overview by billable only - it should show only billables in the table
4. filter by project - it should only show that project. Total time for that project should match the available entries
5. change month - data in table is updated to correctly reflect the selected month
6. go to present - should show you the entire present month
7. print button - should trigger window.print, the layout should be print-friendly, with elements that only appear in print
8. Navigate back to the "/" - shows the current week, not the week you came from

## Screenshots
image
image
image

### CODE REVIEW

File: components/app/top-bar.vue [hunk 1]
```diff
@@ -9,7 +9,7 @@
                 src="@/assets/images/logo-dark.png"
                 alt="frontmen logo"
                 @click="handleLogoClick"
-              />
+              >
```
Comment (mmehmetAliIzci) [inline components/app/top-bar.vue:12]: Is this removed on purpose ?

File: components/records/navigation-buttons.vue [hunk 2]
```diff
@@ -1,6 +1,12 @@
 <template>
   <div class="navigation-buttons">
     <div class="navigation-buttons__container">
+      <nuxt-link v-if="!isAdminView" to="month">
```
Comment (mmehmetAliIzci) [inline components/records/navigation-buttons.vue:4]: Maybe we should show this view for the admins too ?

File: components/reports/month-navigation-buttons.vue [hunk 3]
```diff
@@ -19,12 +19,11 @@
       <h2 class="navigation-buttons__week-label">
         {{ monthLabel }}
       </h2>
+      <b-button v-if="monthDifference !== 0" @click="handleCurrentClick()">
```
Comment (mmehmetAliIzci) [inline components/reports/month-navigation-buttons.vue:22]: Almost embarrassed how nitpick this is but `handleCurrentClick` is sufficient 😬

File: components/records/navigation-buttons.vue (hunk 2 above)
Comment (vladPinteaFrontmen) [reply components/records/navigation-buttons.vue:4]: this refers to /timesheets view of a timesheet vs / view, not admins and regular employees

File: pages/month.vue [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+
+    const selectedCustomers = ref<{value: string; label: string;}[]>([]);
+    const onlyBillable= ref<boolean>(false);
+
+    const employee = computed(() => {
+      return store.state.employee.employee;
+    });
+
+    const getRecords = () => {
+        const startDate = monthDate.value;
```
Comment (mmehmetAliIzci) [inline pages/month.vue:191]: Nitpick: spacing

File: components/app/top-bar.vue (hunk 1 above)
Comment (vladPinteaFrontmen) [reply components/app/top-bar.vue:12]: prettier removed it, but it leads to no conflicts, it's normal for img to just be 1 tag. I think reacts doesn't like img and br without the self-closing slash, but normally html considers it valid and vue seems to not think it's a problem

File: components/reports/month-navigation-buttons.vue (hunk 3 above)
Comment (vladPinteaFrontmen) [reply components/reports/month-navigation-buttons.vue]: done

File: pages/month.vue [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+      () => {
+        endDate = endOfMonth(monthDate.value);
+        getRecords();
+      },
+      {
+        immediate: true,
+      },
+    );
+
+    const monthReport = computed(() => {
```
Comment (mmehmetAliIzci) [inline pages/month.vue:212]: Decide if you want to do this. Might harm readability but i think its oke
`const monthReport = computed(() => handleSelectedProjects(handleFilterBillable(store.state.records.timeRecords)));`

Comment (vladPinteaFrontmen) [reply pages/month.vue:212]: yeah, I decided against that for readability

File: pages/month.vue (hunk 4 above)
Comment (vladPinteaFrontmen) [reply pages/month.vue]: done

**Review by mmehmetAliIzci — APPROVED** [review]

**Comment by mmehmetAliIzci** [comment]
Looks awesome 💯

**Pull request merged**